### ISSUE TITLE
ECSO release

### ISSUE THREAD

**Issue by stevenchong** [DataONEorg/sem-prov-ontologies#41]
We can use this issue to keep track of the steps needed to proceed towards the upcoming ECSO release. I'm assuming we shouldn't do any major edits until after the first release, so I'll keep track of any needed future changes as Github issues.

**Comment by amoeba** [comment]
Hey @stevenchong, what kind of release are we thinking of targeting? ECSO release management has been pretty ad-hoc so far, it appears. Informally, ECSO is currently at 0.9, though that's not exactly codified anywhere. If this release is iterative, we could do a 0.10.

Or do you know if the goal is to get to a 1.0 (first) release state with this effort?

**Comment by amoeba** [comment]
Added a few items as a checklist to the OP.

**Comment by mbjones** [comment]
What version of ECSO are we using for the already production semantics release?  I assume that was well-versioned?  It seems to be 0.9 (http://bioportal.bioontology.org/ontologies/ECSO/?p=summary) but it also looks like there are different versions on bioportal all with the 0.9 label version number.  Whatever we decide to do, I'd like to be sure that each release version has one and only one definitive owl file associated with it.

**Comment by amoeba** [comment]
Yeah, it's "0.9". BioPortal handles releases a little wonky so there are unintended duplicates there. 0.9 wasn't released in the proper way since we couldn't get things merged to master in time (and we're still working on that).

> I'd like to be sure that each release version has one and only one definitive owl file associated with it.

Me too. This will fall out of the above process.

**Comment by stevenchong** [comment]
>  what kind of release are we thinking of targeting?

Good question. I know there are some term reorganization and label issues that we would like to resolve. A 1.0 release will require a thorough review of ECSO by @mpsaloha , which probably won't happen for a while. I think minor improvement updates are okay until that happens. Eventually we will want to add more relationships (axioms) to terms, which seems like a significant update to me.

**Comment by amoeba** [comment]
Sounds like a 0.10 is in order. Thanks @stevenchong. What do you think about making our next semantics call a time to talk over checklist item 2 (triage)?

**Comment by stevenchong** [comment]
Sounds good to me

**Comment by mbjones** [comment]
@amoeba Let's be sure to capture the current 0.9 release of ECSO in the repo and tag it.  I will create a branch for this new ecso 0.10 feature work.

**Comment by mbjones** [comment]
To follow the new contributing guidelines in #89, I created a new branch `feature_41_ecso_0.10` to contain the work on the ECSO 0.10 release under this ticket. This new branch was cloned from the branch `ECSO8-add_non-carbon_measurements` to get what I think is the current checked in version of ECSO8.owl, and then changes were merged into it from `develop` to catch it up with housekeeping changes. Finally, I renamed the `observation` directory to `ecso` to make it clear which ontology is contained within.  All work to resolve issues for the ECSO 0.10 should happen on this new `feature_41_ecso_0.10` branch.  We should also git rm any old copies of the ECSO file that are around in this branch (e.g., ECSO4.owl, ECSO8.owl) in favor of a standard ECSO.owl filename that would reflect the merged consensus for the release. Having multiple copies under various names is confusing -- I hope we can use git to handle the versioning, not filesystem copies.

@amoeba please review if I got the correct versions from the historical branches to start from. Thanks.

**Comment by amoeba** [comment]
Hey @mbjones, thanks for going through the branches. That all matches my memory so I think we're good. The next release of ECSO was waiting on @mpsaloha for a set of eyes so we might touch base with him to see if he has any changes or if we can cut a release here. I'll put it on the agenda for this week's salmantics call.

**Comment by amoeba** [comment]
As a note, we did have a conversation about the numerous branches and that thread is at https://github.com/DataONEorg/sem-prov-ontologies/issues/85. Looks like the conclusion is that we're on the right track and ECSO8-add_non-carbon_measurements is the latest.
